support/datastore/gcs_datastore.go

```
@@ -103,3 +106,43 @@ func (b *GCSDataStore) putFile(ctx context.Context, filePath string, in io.Write
 	}
 	return w.Close()
 }
+
+func (b *GCSDataStore) ListDirectoryNames(ctx context.Context) ([]string, error) {
+	var directories []string
+
+	o := b.bucket.Objects(ctx, &storage.Query{Prefix: b.prefix + "/", Delimiter: "/"})
+	for {
+		attrs, err := o.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatal(err)
```

why not return the err?

---

You're right it should just return the err